test/tests/carousel_carousel-2-tablist.js

```diff
@@ -0,0 +1,450 @@
+      'right arrow on tab "' + index + '" should put focus on the next tab.'
+    );
+
+    t.true(
+      await waitAndCheckAriaSelected(t, index + 1),
+      'right arrow on tab "' + index + '" should set aria-selected="true" on next tab.'
+    );
+    t.true(
+      await tabpanels[index + 1].isDisplayed(),
+      'right arrow on tab "' + index + '" should display the next tab panel.'
+    );
+  }
+
+  // Send the arrow right key to move focus
+  await tabs[tabs.length - 1].sendKeys(Key.ARROW_RIGHT);
+
+  // Check the focus returns to the first item
+  t.true(
+    await waitAndCheckFocus(t, ex.tabSelector, 0),
+    'right arrow on tab "' + (tabs.length - 1) + '" should put focus to first tab.'
+  );
+  t.true(
+    await waitAndCheckAriaSelected(t, 0),
+    'right arrow on tab "' + (tabs.length - 1) + '" should set aria-selected="true" on first tab.'
+  );
+  t.true(
+    await tabpanels[0].isDisplayed(),
+    'right arrow on tab "' + (tabs.length - 1) + '" should display the first panel.'
+  );
+
+});
+
+
+ariaTest('ARROW_LEFT key moves focus and activates tab', exampleFile, 'key-left-arrow', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+
+  // Put focus on first tab
+  await openTabAtIndex(t, 0);
+
+  // Send the left arrow
+  await tabs[0].sendKeys(Key.ARROW_LEFT);
+
+  // Check the focus returns to the last item
+  t.true(
+    await waitAndCheckFocus(t, ex.tabSelector, tabs.length - 1),
+    'left arrow on tab 0 should put focus to last tab.'
+  );
+
+  t.true(
+    await waitAndCheckAriaSelected(t, tabs.length - 1),
+    'left arrow on tab 0 should set aria-selected="true" on last tab.'
+  );
+  t.true(
+    await tabpanels[tabs.length - 1].isDisplayed(),
+    'left arrow on tab 0 should display the last panel.'
+  );
+
+  for (let index = tabs.length - 1; index > 0; index--) {
+
+    // Send the arrow left key to move focus
+    await tabs[index].sendKeys(Key.ARROW_LEFT);
+
+    // Check the focus is correct
+    t.true(
+      await waitAndCheckFocus(t, ex.tabSelector, index - 1),
+      'left arrow on tab "' + index + '" should put focus on the previous tab.'
+    );
+    t.true(
+      await waitAndCheckAriaSelected(t, index - 1),
+      'left arrow on tab "' + index + '" should set aria-selected="true" on previous tab.'
+    );
+    t.true(
+      await tabpanels[index - 1].isDisplayed(),
+      'left arrow on tab "' + index + '" should display the next previous panel.'
+    );
+  }
+});
+
+ariaTest('HOME key moves focus and selects tab', exampleFile, 'key-home', async (t) => {
+  t.plan(3 * ex.tabCount);
+
+  const tabs = await t.context.session.findElements(By.css(ex.tabSelector));
+  const tabpanels = await t.context.session.findElements(By.css(ex.tabpanelSelector));
+  for (let index = 0; index < tabs.length; index++) {
```

Comment:
The home key logic is independent of which tab has focus -- would it make sense to have two tests here instead, one with focus on the first tab, and one with focus on any other tab?

Reply:
that is a reasonable, but this test is also otherwise complete so I think it's fine to leave as is!